### DISCUSSION TITLE
Not a recognised date format

### DISCUSSION THREAD

**Discussion by poudenes** [firefly-iii/firefly-iii#6949] [Firefly III]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [X] ... it's [definitely me, not you](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

I changed for Firefly and Firefly importer the IP address inside docker. Before it worked great. After changing this I see errors. All URL path are also updated for Firefly importer so it knows the new IP address

### Expected behaviour

Only change IP and update all settings with new IP must work normal without errors

### Debug information

**FIREFLY IMPORTER: V1.0.2**
```
[2023-01-31 10:34:17] production.ERROR: Status code is 500: {"message":"Internal Firefly III Exception: [d]Not a recognised date format: \"120\"","exception":"FireflyIII\\Exceptions\\FireflyException"}  
[2023-01-31 10:34:17] production.DEBUG: Add error on index #1 (line no. 2): Submission HTTP error: Status code is 500: {&quot;message&quot;:&quot;Internal Firefly III Exception: [d]Not a recognised date format: \&quot;120\&quot;&quot;,&quot;exception&quot;:&quot;FireflyIII\\Exceptions\\FireflyException&quot;} 
```

**FIREFLY: V 5.7.18**
```
[2023-01-31 10:34:19] local.ERROR: Exception is: {"class":"FireflyIII\\Exceptions\\FireflyException","errorMessage":"[d]Not a recognised date format: \"120\"","time":"Tue, 31 Jan 2023 10:34:19 +0100","file":"\/var\/www\/html\/app\/Support\/ParseDateString.php","line":121,"code":0,"version":"5.7.18","url":"http:\/\/172.30.33.119:8080\/api\/v1\/transactions","userAgent":"GuzzleHttp\/7","json":true,"method":"POST"}  
[2023-01-31 10:34:19] local.ERROR: [d]Not a recognised date format: "120" {"userId":1,"exception":"[object] (FireflyIII\\Exceptions\\FireflyException(code: 0): [d]Not a recognised date format: \"120\" at /var/www/html/app/Support/ParseDateString.php:121)
```

### Steps to reproduce

1. Change IP in Portainer Stack and update container
2. Update firefly_importer stack with new IP and update container
3. Firefly itself is working normal when access UI
4. Strange is that all imports are in Firefly

**Comment by JC5** [comment]
Since you're POST-ing transactions, the error comes from whatever you're submitting.

**Reply by poudenes** [reply]
> Since you're POST-ing transactions, the error comes from whatever you're submitting.

Only thing that I do is automatic Firefly-importer and sometimes manual import using the firefly-importer.
Nothing else.

Tried is later again and didn't see the error at that moment.
Ill keep a eye on this.

**Comment by poudenes** [comment]
Issue still there. I only use the auto import that I use for months now. All data is coming from nordigen.

Import is done. Rules in Firefly are not fired after that. Must do it manual

This is for 2 accounts with Rabobank and ING

**Reply by JC5** [reply]
I'll need some logs to see what's going on. Could you share the logs with me when you run into a faulty transaction? I'm curious to see what the importer tries to submit.

**Comment by poudenes** [comment]
I can send you a great log. Will mail it to you private. It's not easy to post it here. Can you give me a email address?

**Reply by JC5** [reply]
james@firefly-iii.org 👍🏻

**Comment by JC5** [comment]
Thanks! All I can tell from this is that there is probably a rule with a bad time range or date stamp. You could share the Firefly III logs with me, or disable any rules with date ranges in them.

**Reply by poudenes** [reply]
> Thanks! All I can tell from this is that there is probably a rule with a bad time range or date stamp. You could share the Firefly III logs with me, or disable any rules with date ranges in them.

And you mean with time or date range is all rules that have a trigger with based on time or date?
If so, then I don't have any rules based on that.

I'll will do some tests on Monday. Will send some logs then also!

**Reply by JC5** [reply]
Exactly! The error comes from the search engine, so I am lead to believe that a rule would have triggered it. But I could be wrong. I appreciate your efforts!

**Comment by poudenes** [comment]
As we speak I maybe found the issue.
In many rules for contracts I added "Set budget to" because in beginning of using Firefly I created this to manage my expenses on contracts as well. 

Some days ago I removed this budget. Because I see in contact list all payments that has been done already.

Many rules had this Set budget to still in there rule list. I removed all those lines because this specific budget doesn't exist anymore.

Sounds logic that Firefly doesn't know what to do then?

**Reply by JC5** [reply]
That could be the case. I remember the rules *should* ignore these budgets, but I'm curious to see what will happen.

**Comment by poudenes** [comment]
This morning did a manual import. I didn't see the error in the log file. It was just 1 import. So I'll keep it in scope for couple of days, weeks. But this seems to be ok

**Comment by poudenes** [comment]
It seems to work normal again. Also the rules are fired again. So Maybe the ignore part of budgets was the problem?

**Reply by JC5** [reply]
That could be it! I'll keep an eye out.
